replication/backup.go

```diff
@@ -13,7 +13,39 @@ import (
 
 // StartBackup: Like mysqlbinlog remote raw backup
 // Backup remote binlog from position (filename, offset) and write in backupDir
-func (b *BinlogSyncer) StartBackup(backupDir string, p Position, timeout time.Duration) error {
+func (b *BinlogSyncer) StartBackupToFile(backupDir string, p Position, timeout time.Duration) error {
+	err := os.MkdirAll(backupDir, 0755)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	return b.StartBackup(p, timeout, func(filename string) (io.WriteCloser, error) {
+		return os.OpenFile(path.Join(backupDir, filename), os.O_CREATE|os.O_WRONLY, 0644)
+	})
+}
+
+// StartBackup initiates a backup process for binlog events starting from a specified position.
+// It continuously fetches binlog events and writes them to files managed by a provided handler function.
+// The backup process can be controlled with a timeout duration, after which the backup will stop if no new events are received.
+//
+// Parameters:
+//   - p Position: The starting position in the binlog from which to begin the backup.
+//   - timeout time.Duration: The maximum duration to wait for new binlog events before stopping the backup process.
+//     If set to 0, a default very long timeout (30 days) is used instead.
+//   - handler func(filename string) (io.WriteCloser, error): A function provided by the caller to handle file creation and writing.
+//     This function is expected to return an io.WriteCloser for the specified filename, which will be used to write binlog events.
+//
+// The function first checks if a timeout is specified, setting a default if not. It then enables raw mode parsing for binlog events
```

Comment:
From this paragraph till the end of comments, they are explain the implementation of the function, I think it's better to move them inside the function (maybe the first line in the function body?), rather than stay here as a function-level comment.

Just my own opinion, I want to see a short explanaion, the meaning of arguments and return values, calling conventions and short examples of the function in function-level comments. If I want to check the implementation details, I can go inside the function to see source codes and more context-related comments